pkg/component/worker/nllb/envoy.go

```diff
@@ -141,11 +144,18 @@ func (e *envoyProxy) start(ctx context.Context, profile workerconfig.Profile, ap
 	}
 
 	nllb := profile.NodeLocalLoadBalancing
+	var konnectivityBindPort *uint16
```

Comment:
I'd remove the `konnectivityServerBindPort` on line 88. No need to have it twice. That way, the below if statement doesn't need to be duplicated.

Reply:
there seems to be some expectation, that the default value of the port is 0. should I change the type of the params value to uint16 from *uint16 and make it 0 instead of nil?

Reply:
I did both changes as separate commits, I can squash that however you think is best.